swan/scripts/before-notebook.d/01_storage.sh

```diff
@@ -23,7 +23,7 @@ then
     chmod 600 $OAUTH2_FILE
 fi
 
-sudo -E -u $NB_USER sh -c 'if [[ ! -d "$HOME" || ! -x "$HOME" ]]; then exit 1; fi'
+run_as_user 'if [[ ! -d "$HOME" || ! -x "$HOME" ]]; then exit 1; fi'
```

Comment:
In which one of the three cases here we were already running as user?

Also, when encapsulating inside a function, do we preserve the exit code (checked below) of the last command executed inside the function?

Reply:
We start the session as root, but the image allows us to start directly as the user. E.g. when running locally.
I did this fix because the image is broken for anything except when running from the SWAN spawner.

Reply:
As for the exit code, I need to confirm.

Reply:
So, the exit code is taken, so all is good.

 But while testing this, I realised that we would never reach the if below that line... Because we instruct bash to exit on failure, doing `exit 1` would make the startup exit then.
I added 1 more commit to allow us to reach the log message.